goinstant/godogs_test.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+
+	"github.com/cucumber/godog"
+	"github.com/pkg/errors"
+)
+
+var (
+	binaryFilePath string
+	logs           string
+	directoryNames []string
+)
+
+func theCommandIsRun(command string) error {
+	res, err := runTestCommand(binaryFilePath, strings.Split(command, " ")...)
+	if err == nil {
+		logs = res
+	}
+	return nil
+}
+
+func checkTheCLIOutputIs(expectedOutput string) error {
+	err := compareStrings(logs, expectedOutput)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func checkCustomPackages(packages *godog.Table) error {
+	head := packages.Rows[0].Cells
+
+	for i := 1; i < len(packages.Rows); i++ {
+
+		for n, cell := range packages.Rows[i].Cells {
+			switch head[n].Value {
+			case "directory":
+				directoryNames = append(directoryNames, cell.Value)
+			case "location":
+				err := compareStrings(logs, cell.Value)
+				if err != nil {
+					return err
+				}
+			default:
+				return errors.New("Unexpected column name: " + head[n].Value)
+			}
+		}
+	}
+
+	// Cleanup downloaded packages post test
+	deleteContentAtFilePath([]string{".", "features"}, directoryNames)
+	return nil
+}
+
+func compareStrings(inputLogs, expectedOutput string) error {
+	if !strings.Contains(inputLogs, expectedOutput) {
+		return errors.New("Logs received: '" + inputLogs + "\nSubstring expected: " + expectedOutput)
+	}
+	return nil
+}
+
+func InitializeScenario(sc *godog.ScenarioContext) {
+	suite := &godog.TestSuite{
+		TestSuiteInitializer: func(s *godog.TestSuiteContext) {
+			s.AfterSuite(cleanUp)
+		},
+		ScenarioInitializer: func(sc *godog.ScenarioContext) {
+			if binaryFilePath == "" {
+				binaryFilePath = buildBinary()
+			}
+			sc.Step(`^check the CLI output is "([^"]*)"$`, checkTheCLIOutputIs)
+			sc.Step(`^the command "([^"]*)" is run$`, theCommandIsRun)
+			sc.Step(`^check that the CLI added custom packages$`, checkCustomPackages)
+		},
+	}
+
+	if suite.Run() != 0 {
+		fmt.Println("Tests failed")
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+func buildBinary() string {
+	_, err := runTestCommand("/bin/sh", filepath.Join(".", "features", "build-cli.sh"))
+	if err != nil {
+		panic(err)
+	}
+	_, err = os.Stat(filepath.Join(".", "features", "test-platform-linux"))
```

Comment:
Why are we checking a specific OS binary file here? Is this just to check if the `build-cli.sh` script succeeded?

Reply:
Yeah, its for checking whether `build.cli` has succeeded